### PR TITLE
Da/runtime extensions

### PR DESCRIPTION
**Pull request by tadad**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -98,6 +103,9 @@ async fn main() {
     // encryptor handles end-to-end encryption for client messages
     let (encryptor_sender, encryptor_receiver): (MessageSender, MessageReceiver) =
         mpsc::channel(ENCRYPTOR_CHANNEL_CAPACITY);
+    // optional llm extension
+    let (llm_sender, llm_receiver): (MessageSender, MessageReceiver) =
+        mpsc::channel(LLM_CHANNEL_CAPACITY);
```
Comment (dr-frmr) [inline src/main.rs:107]: should this be masked under a `cfg` line too?

File: src/main.rs [hunk 2]
```diff
@@ -30,6 +34,7 @@ const ETH_RPC_CHANNEL_CAPACITY: usize = 32;
 const VFS_CHANNEL_CAPACITY: usize = 1_000;
 const ENCRYPTOR_CHANNEL_CAPACITY: usize = 32;
 const CAP_CHANNEL_CAPACITY: usize = 1_000;
+const LLM_CHANNEL_CAPACITY: usize = 32;
```
Comment (dr-frmr) [inline src/main.rs:38]: should this be under a `cfg` line?

File: src/types.rs [hunk 3]
```diff
@@ -1060,3 +1060,90 @@ pub enum EncryptorMessage {
...
 }
 // encryptor End
+
+// llm.rs Start
+#[derive(Debug, Serialize, Deserialize)]
+pub struct LlmPrompt {
+    pub prompt: String,
+    pub n_predict: u64,
+}
+
```
Comment (dr-frmr) [inline src/types.rs]: these types should be in their own file, preferably under a folder named `llm` that contains all the code related to the llm extension

**Review by dr-frmr — COMMENTED** [review]
nice code reuse across FS and kernel, just a few notes on keeping the llm feature firewalled and separate

File: src/types.rs (hunk 3 above)
Comment (tadad) [reply src/types.rs]: true, moved into `/llm/types.rs`

**Review by dr-frmr — APPROVED** [review]

**Pull request merged**